selftest/test_images.py

```diff
@@ -0,0 +1,332 @@
+]
+
+DEFAULT_ARM64_VM_SIZES = [
+    sku.vm_size for sku in selftest.SKU_CONFIGS.values() if sku.vm_size_type == "arm64"
+]
+
+CUSTOM_IMAGES = [
+    image for image in os.getenv("TEST_CUSTOM_IMAGES", "").split(",") if image
+]
+ENV_TEST_CUSTOM_VM_SIZES = os.getenv("TEST_CUSTOM_VM_SIZES", "")
+if ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_GEN1_VM_SIZES":
+    CUSTOM_VM_SIZES = DEFAULT_GEN1_VM_SIZES
+elif ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_GEN2_VM_SIZES":
+    CUSTOM_VM_SIZES = DEFAULT_GEN2_VM_SIZES
+elif ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_ARM64_VM_SIZES":
+    CUSTOM_VM_SIZES = DEFAULT_ARM64_VM_SIZES
+else:
+    CUSTOM_VM_SIZES = [image for image in ENV_TEST_CUSTOM_VM_SIZES.split(",") if image]
+
+
+def subprocess_run(cmd: List[str], check: bool = True):
+    """Run a subprocess command and capture outputs as utf-8."""
+    print(f"executing command: {shlex.join(cmd)}")
+    try:
+        proc = subprocess.run(
+            cmd,
+            check=check,
+            capture_output=True,
+            text=True,
+            encoding="utf-8",
+        )
+    except subprocess.CalledProcessError as error:
+        print(
+            f"error running command: {error} stdout={error.stdout} stderr={error.stderr}"
+        )
+        raise
+
+    print(f"executed command: {proc}")
+    return proc
+
+
+@pytest.fixture
+def ssh_key_path(tmp_path):
+    """Generate a temporary SSH key pair for the test."""
+    path = tmp_path / "id_rsa"
+    subprocess.run(
+        ["ssh-keygen", "-t", "rsa", "-b", "2048", "-f", str(path), "-N", ""], check=True
+    )
+    yield path
+
+
+@pytest.fixture
+def azure_subscription():
+    """Get the Azure subscription ID."""
+    subscription = os.getenv("AZURE_SUBSCRIPTION_ID") or ""
+    assert subscription, "AZURE_SUBSCRIPTION_ID environment variable is required"
+    yield subscription
+
+
+@pytest.fixture
+def azure_location():
+    """Get the Azure location."""
+    location = os.getenv("AZURE_LOCATION") or ""
+    assert location, "AZURE_LOCATION environment variable is required"
+    yield location
+
+
+@pytest.fixture
+def temp_resource_group(azure_subscription, azure_location):
+    """Create a temporary resource group for the test."""
+    resource_group_name = f"test-rg-{uuid.uuid4()}"
+    delete_after = (datetime.utcnow() + timedelta(hours=1)).isoformat()
+    subprocess.run(
+        [
+            "az",
+            "group",
+            "create",
+            "--subscription",
+            azure_subscription,
+            "--location",
+            azure_location,
+            "--name",
+            resource_group_name,
+            "--tags",
+            f"deleteAfter={delete_after}",
+        ],
+        check=True,
+    )
+    yield resource_group_name
+    subprocess.run(
+        ["az", "group", "delete", "--name", resource_group_name, "--yes", "--no-wait"],
+        check=True,
+    )
+
+
+class TestVMs:
+    """Test VMs with different images and sizes."""
+
+    @pytest.fixture(autouse=True)
```

Comment:
NIT: I think autouse is actually the default behavior so long as you provide the name of the fixture as the parameter into the test

Reply:
yeah I suppose that works too. I didn't name the fixture as the autouse applies to the tests in the class